target/linux/lantiq/patches-5.10/0702-net-lantiq-enable-baby-jumbo-frames-on-xrx200.patch

```diff
@@ -0,0 +1,125 @@
+From d39d31486432bb3e54e4cb83744d99c3f5d773c6 Mon Sep 17 00:00:00 2001
+From: Thomas Nixon <tom@tomn.co.uk>
+Date: Sat, 3 Jul 2021 15:57:03 +0100
+Subject: [PATCH] net: lantiq: enable baby jumbo frames on xrx200
+
+xrx200 max MTU is reduced so that it works correctly when set to the
+max, and the max MTU of the switch is increased.
+
+The switch driver now enables non-standard MTUs on a per-port basis,
+with the overall frame size set based on the cpu port.
+
+When the MTU is not changed, this should have no effect.
+
+Signed-off-by: Thomas Nixon <tom@tomn.co.uk>
+---
+ drivers/net/dsa/lantiq_gswip.c       | 42 ++++++++++++++++++++++++++--
+ drivers/net/ethernet/lantiq_xrx200.c |  4 ++-
+ 2 files changed, 42 insertions(+), 4 deletions(-)
+
+diff --git a/drivers/net/dsa/lantiq_gswip.c b/drivers/net/dsa/lantiq_gswip.c
+index 93c7fa1fd4cb..1e67562153f9 100644
+--- a/drivers/net/dsa/lantiq_gswip.c
++++ b/drivers/net/dsa/lantiq_gswip.c
+@@ -238,6 +238,8 @@
+ 
+ #define XRX200_GPHY_FW_ALIGN	(16 * 1024)
+ 
++#define GSWIP_MAX_PACKET_LENGTH	9600
```

Comment:
The PHYs can handle up to 9990 bytes and the switch buffer manager up to 10240 bytes.
We should make sure to not send packets bigger than 9990 bytes.

Reply:
Sorry for the delay. The 9600 value came from the DM200 GPL dump; I could increase this, but as written above I  can't test these larger sizes.